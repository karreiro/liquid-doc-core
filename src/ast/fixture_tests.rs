// In your test file
include!(concat!(env!("OUT_DIR"), "/fixtures_macro.rs"));

macro_rules! generate_fixture_test {
    ($fixture_name:expr) => {
        paste::paste! {
            #[test]
            fn [<test_fixture_ $fixture_name>]() {
                let content = include_str!(concat!("../../web/fixtures/", $fixture_name, ".liquid"));
                // Your test logic here
                crate::assert_json_output!(content);
            }
        }
    };
}

// This will generate all the tests
for_each_fixture_file!(generate_fixture_test);
