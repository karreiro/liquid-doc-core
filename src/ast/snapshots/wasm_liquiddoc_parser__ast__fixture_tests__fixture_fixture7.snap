---
source: src/ast/fixture_tests.rs
expression: "this is an implicit description\nin a header\n\n@description with a description annotation"
---
[
  {
    "type": "LiquidDocDescriptionNode",
    "name": "description",
    "position": {
      "start": 10,
      "end": 55
    },
    "source": "this is an implicit description\nin a header\n\n",
    "content": {
      "type": "TextNode",
      "value": "this is an implicit description\nin a header\n\n",
      "position": {
        "start": 10,
        "end": 55
      },
      "source": "this is an implicit description\nin a header\n\n"
    },
    "isImplicit": true,
    "isInline": true
  },
  {
    "type": "LiquidDocDescriptionNode",
    "name": "description",
    "position": {
      "start": 55,
      "end": 98
    },
    "source": "@description with a description annotation",
    "content": {
      "type": "TextNode",
      "value": "with a description annotation\n",
      "position": {
        "start": 68,
        "end": 98
      },
      "source": "@description with a description annotation"
    },
    "isImplicit": false,
    "isInline": true
  }
]
